src/renderer/contexts/GlobalNodeState.tsx

```diff
@@ -800,6 +816,37 @@ export const GlobalProvider = ({
         [modifyNode]
     );
 
+    useHotkeys(
+        'ctrl+c, cmd+c',
+        () => {
+            const selectedNodes = getNodes().filter((n) => n.selected);
+            const selectedEdges = getEdges().filter((e) => e.selected);
+            const copyData = Buffer.from(
+                JSON.stringify({ nodes: selectedNodes, edges: selectedEdges })
+            );
+            clipboard.writeBuffer('application/chainner.chain', copyData, 'clipboard');
+        },
+        [getNodes]
+    );
+
+    useHotkeys(
+        'ctrl+v, cmd+v',
+        () => {
+            const { nodes, edges } = JSON.parse(
+                clipboard.readBuffer('application/chainner.chain').toString()
+            ) as {
+                nodes: Node[];
+                edges: Edge[];
+            };
+            if (nodes.length > 0) {
+                const nodesToCopy = new Set(nodes.map((n) => n.id));
+                const edgesToCopy = edges.length > 0 ? new Set(edges.map((e) => e.id)) : null;
+                copyNodesAndEdges(nodesToCopy, edgesToCopy);
+            }
+        },
+        [copyNodesAndEdges]
+    );
```

Comment:
That's not a good way of making sure types align. Right now, if you changed this type to have one more field, or you removed a field from `writeBuffer` above, then would you not get a type error.

It's better to make an interface to make sure that both the write and read have no choice but to follow the interface.

```suggestion
interface ClipboardChain {
                nodes: Node[];
                edges: Edge[];
            }
            
    useHotkeys(
        'ctrl+c, cmd+c',
        () => {
            const selectedNodes = getNodes().filter((n) => n.selected);
            const selectedEdges = getEdges().filter((e) => e.selected);
            cosnt data: ClipboardChain = { nodes: selectedNodes, edges: selectedEdges }
            const copyData = Buffer.from(
                JSON.stringify(data)
            );
            clipboard.writeBuffer('application/chainner.chain', copyData, 'clipboard');
        },
        [getNodes]
    );

    useHotkeys(
        'ctrl+v, cmd+v',
        () => {
            const { nodes, edges } = JSON.parse(
                clipboard.readBuffer('application/chainner.chain').toString()
            ) as ClipboardChain;
            if (nodes.length > 0) {
                const nodesToCopy = new Set(nodes.map((n) => n.id));
                const edgesToCopy = edges.length > 0 ? new Set(edges.map((e) => e.id)) : null;
                copyNodesAndEdges(nodesToCopy, edgesToCopy);
            }
        },
        [copyNodesAndEdges]
    );
```